specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/rbac.json

```diff
@@ -0,0 +1,566 @@
+            "schema": {
+              "$ref": "#/definitions/SqlRoleAssignmentListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SqlRoleDefinitionResource": {
+      "properties": {
+        "roleName": {
+          "type": "string",
+          "description": "A user-friendly name for the Role Definition. Must be unique for the database account."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "BuiltInRole",
+            "CustomRole"
+          ],
+          "description": "Indicates whether the Role Definition was built-in or user created.",
+          "x-ms-enum": {
+            "name": "RoleDefinitionType",
+            "modelAsString": false
+          }
+        },
+        "assignableScopes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A set of fully qualified Scopes at or below which Role Assignments may be created using this Role Definition. This will allow application of this Role Definition on the entire database account or any underlying Database / Collection. Must have at least one element. Scopes higher than Database account are not enforceable as assignable Scopes. Note that resources referenced in assignable Scopes need not exist."
+        },
+        "permissions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Permission"
+          },
+          "description": "The set of operations allowed through this Role Definition."
+        }
+      },
+      "description": "Azure Cosmos DB SQL Role Definition resource object."
+    },
+    "SqlRoleDefinitionCreateUpdateParameters": {
+      "description": "Parameters to create and update an Azure Cosmos DB SQL Role Definition.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to create and update an Azure Cosmos DB SQL Role Definition.",
+          "$ref": "#/definitions/SqlRoleDefinitionResource"
+        }
+      }
+    },
+    "SqlRoleDefinitionGetResults": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SqlRoleDefinitionResource",
+          "description": "Properties related to the Role Definition."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "cosmos-db.json#/definitions/ARMProxyResource"
+        }
+      ],
+      "description": "An Azure Cosmos DB SQL Role Definition."
+    },
+    "Permission": {
+      "type": "object",
+      "properties": {
+        "dataActions": {
```

Comment:
Let's not remove notDataActions entirely. RP still returns it as an empty array. We simply don't allow users to add to it.